### ISSUE TITLE
md-tab causes $digest already in progress error

### ISSUE THREAD

**Issue by ismarslomic** [angular/material#2850]
When switching the tabs in md-tab directive I get error message below.

Dependencies:
- Angular v1.4.0-rc.2
- Angular Material (master branch)

Codepen: http://codepen.io/ismarslomic/pen/waGjwQ
Error message displayed in Developer tools console:
```
Error: [$rootScope:inprog] $digest already in progress
http://errors.angularjs.org/1.4.0-rc.2/$rootScope/inprog?p0=%24digest
    at REGEX_STRING_REGEXP (angular.js:68)
    at beginPhase (angular.js:16056)
    at Scope.$get.Scope.$apply (angular.js:15800)
    at Scope.scopePrototype.$apply (hint.js:1478)
    at HTMLElement.<anonymous> (hint.js:797)
    at HTMLElement.eventHandler (angular.js:3247)
    at redirectFocus (angular-material.js:12919)
    at Object.handleFocusIndexChange [as fn] (angular-material.js:12915)
    at Scope.$get.Scope.$digest (angular.js:15539)
    at Scope.scopePrototype.$digest (hint.js:1468)
´´´

**Comment by erikdonohoo** [comment]
What browser are you using?  On Mac OS 10.10 with Chrome 42 I see no error.

**Comment by ismarslomic** [comment]
Mac OSX 10.10.3, Chrome 42.0.2311.135 (64-bit).

I neither se the error when running the code in Codepen. But if you put the same code in index.html file and just open it locally you will se the error. The samme happens when the code is hosted by NodeJS for instance.

Complete index.html:
```
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Angular Material Tabs</title>
    <link rel="stylesheet" href="https://rawgit.com/angular/bower-material/master/angular-material.css">
    <base href="/">
</head>

<body ng-app="mdTabsApp" layout="column">
<md-content class="md-padding" flex layout="row" layout-align="center start" style="background-color: #eee;">
    <md-tabs md-stretch-tabs="always"
             md-border-bottom
             class="md-whiteframe-z2"
             flex
             style="background-color:#ffffff">
        <md-tab>
            <md-tab-label>Tab 1</md-tab-label>
            <md-tab-body class="md-padding">
                Tab 1 content
            </md-tab-body>
        </md-tab>
        <md-tab>
            <md-tab-label>Tab 2</md-tab-label>
            <md-tab-body class="md-padding">
                Tab 2 content
            </md-tab-body>
        </md-tab>
        <md-tab>
            <md-tab-label>Tab 3</md-tab-label>
            <md-tab-body class="md-padding" >
                Tab 3 content
            </md-tab-body>
        </md-tab>
    </md-tabs>
</md-content>

<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.0-rc.2/angular.js"></script>
<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.0-rc.2/angular-animate.js"></script>
<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.0-rc.2/angular-aria.js"></script>
<script src="https://rawgit.com/angular/bower-material/master/angular-material.js"></script>

<script>
    (function (angular, undefined) {
        "use strict";
        angular.module('mdTabsApp', [
            'ngMaterial'
        ])
    })(angular);
</script>

</body>
</html>

**Comment by tony-cocco** [comment]
@erikdonohoo 

Because I've been inspecting this on [stackoverflow](http://stackoverflow.com/questions/30243389/md-tab-causes-digest-already-in-progress-error), I made a [runnable](http://runnable.com/VVTltKLzNt5xHAEt/md-tab-causes-%24digest-already-in-progress-error-for-express-and-node-js) and couldn't reproduce this.

Firefox 31 on Windows 7.

**Comment by ismarslomic** [comment]
Could you try with Chrome And hosting with nodejs/expressjs?

**Comment by tony-cocco** [comment]
@ismarslomic  

The runnable **is** using nodejs/express. With the arguably the exact same express script (the runnable is using port 80 instead of 5000). 

You're more than welcome to run it using Chrome, or any browser you want (except IE, apparently it doesn't work with runnable). Just select "Run". If you really want, you can click the run options and set "Web Only".

**Comment by ismarslomic** [comment]
This is very very strange behaviour. Like I already mentioned, I don´t see this error in Developer tools console when running the index.html file directly, or by using Codepen/Runnable for instance, only when hosting on ExpressJS. And I neither see it in Firefox.. However, you can close this issue as it is most probably something strange with my Chrome installation..

**Comment by tony-cocco** [comment]
Why would express interfere with client-side scripts? Unless those scripts were being mangled before being served. Even then, your HTML (in its posted form) is fetching them, not the server.

**Comment by ismarslomic** [comment]
I see your point, @tony-cocco! And given the app.js file, you see that there are no manipulation of the client side scripts/static files done in express. I assume this has something to do with my local Chrome setup/extensions.

**Comment by ThomasBurleson** [comment]
Unable to reproduce.

**Comment by nitin7dc** [comment]
Getting this on Chrome Version 42.0.2311.152 m (64-bit).

**Comment by ThomasBurleson** [comment]
@nitin7dc - Do you get the same with the latest 0.9.4 release ?

**Comment by nitin7dc** [comment]
@ThomasBurleson  : yes getting on 0.9.4 as well.

**Comment by ThomasBurleson** [comment]
@nitin7dc - Can your provide more details?

*  How did you install Angular-Material (npm, bower, etc)
*  What device (desktop, mobile)?
*  What OS ?
*  Do you have a CodePen or Plunkr that reproduces this online ?

**Comment by ismarslomic** [comment]
@nitin7dc : do you also see this error only when hosting on ExpressJS and not for instance in Codepen or Runnable?

**Comment by ismarslomic** [comment]
Additional comment: expressjs running locally

**Comment by nitin7dc** [comment]
@ThomasBurleson  
How did you install Angular-Material (npm, bower, etc) : bower
What device (desktop, mobile)? : Chrome Version 42.0.2311.152 m (64-bit) (desktop)
What OS ? : Windows 7 Service Pack1 ( 64bit)
Do you have a CodePen or Plunkr that reproduces this online ? Works fine on mozilla, chrome canary, plnk.

**Comment by nitin7dc** [comment]
@ismarslomic : yes only with expressjs

**Comment by jameswyse** [comment]
Are you using the Batarang extension for Chrome? I ran in to the exact same error today and disabling it seems to have fixed the problem.

**Comment by nitin7dc** [comment]
@jameswyse yes i'm using batarang extension. Disabling it fixed the issue !

**Comment by ismarslomic** [comment]
Me 2! Thanks!

**Comment by ohohco** [comment]
fuck it
